cmake/thirdparty/SetupMPI.cmake

```diff
@@ -77,6 +77,14 @@ if (ENABLE_FIND_MPI)
         list(APPEND _mpi_libraries ${MPI_Fortran_LIBRARIES})
     endif()
     blt_list_remove_duplicates(TO _mpi_libraries)
+    
+    # Make nvcc work with MPI when compiling RAJA MPI+CUDA nvcc.
+    # It fixes nvcc compilation for options like -pthread.
+    if (ENABLE_CUDA)
+        if (NOT "${_mpi_compile_flags}" STREQUAL "$<$<COMPILE_LANGUAGE:CUDA>:-Xcompiler=${_mpi_compile_flags}>")
```

Comment:
Since we explicitly build up the `_mpi_compile_flags` variable in this file, I don't think this check can ever fail.

The good news is that, following my comment below, I don't think we need the check.